File: csharp/lucians-luscious-lasagna/LuciansLusciousLasagna.cs
class Lasagna
{
    public static readonly int LayerTime = 2;

    public int ExpectedMinutesInOven() => 40;

    public int RemainingMinutesInOven(int elapsed)   => ExpectedMinutesInOven() - elapsed;

    public int PreparationTimeInMinutes(int layers) => LayerTime* layers;

    public int ElapsedTimeInMinutes(int layers, int elapsed) => PreparationTimeInMinutes(layers) + elapsed;
}

File: fsharp/lucians-luscious-lasagna/LuciansLusciousLasagna.fs
module LuciansLusciousLasagna

let expectedMinutesInOven = 40

let remainingMinutesInOven spent = expectedMinutesInOven - spent

let preparationTimeInMinutes layers = 2 * layers

let elapsedTimeInMinutes layers spent = (preparationTimeInMinutes layers) + spent